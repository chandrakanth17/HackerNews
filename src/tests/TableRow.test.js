import React from 'react';
import renderer from 'react-test-renderer';
import { mount, configure } from 'enzyme';
import Feed from '../fixtures/Feed.json';
import TableRowItem from '../pages/Home/Components/TableRow';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const fnClick = jest.fn();

describe('Table Row Item component should render correct', () => {
  it('Given a single feed data table row item will display the row with data', () => {
    const item = renderer
      .create(
        <TableRowItem feed={Feed} upVote={() => ''} hideArticle={() => ''} />
      )
      .toJSON();
    expect(item).toMatchSnapshot();
  });

  it('On Clicking the upvote the click action should happen ', () => {
    const component = mount(
      <TableRowItem feed={Feed} upVote={fnClick} hideArticle={fnClick} />
    );
    component
      .find('button')
      .first()
      .simulate('click');
    expect(fnClick).toHaveBeenCalled();
  });

  it('On clicking hide the click action should happen', () => {
    const component = mount(
      <TableRowItem feed={Feed} upVote={fnClick} hideArticle={fnClick} />
    );
    component
      .find('button')
      .last()
      .simulate('click');
    expect(fnClick).toHaveBeenCalled();
  });
});
