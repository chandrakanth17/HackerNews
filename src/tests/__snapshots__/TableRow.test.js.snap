// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Row Item component should render correct Given a single feed data table row item will display the row with data 1`] = `
<tr
  className="sc-AxgMl hDlfOi"
>
  <td
    className="sc-AxheI cKREbw"
  >
    159
  </td>
  <td
    className="sc-AxheI cKREbw"
  >
    748
  </td>
  <td
    className="sc-AxheI cKREbw"
  >
    <button
      className="sc-Axmtr haBuNv"
      onClick={[Function]}
    >
      â–²
    </button>
  </td>
  <td
    className="sc-AxheI cKREbw"
  >
    Seinfeld Adventure 
    <a
      className="sc-AxmLO ggKaVv"
      href="https://www.seinfeldgame.com/"
    >
       
      (www.seinfeldgame.com) 
       
    </a>
    <span
      className="sc-fzozJi ixACyM"
    >
      <span
        className="sc-fzpans jofvbJ"
      >
        by
      </span>
      tosh 
       
      <span
        className="sc-fzpans jofvbJ"
      >
        in 3 days ago 
      </span>
      <button
        className="sc-fzoLsD hMdAPp"
        onClick={[Function]}
      >
        [ Hide ]
      </button>
    </span>
  </td>
</tr>
`;
